lib/rouge/lexers/json5.rb

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    load_lexer 'javascript.rb'
+
+    class JSON5 < JSON
+      title 'JSON 5'
+      tag 'json5'
+      filenames '*.json5'
+      mimetypes 'application/json5', 'application/x-json5'
+
+      desc 'JSON 5 extension for JSON (json5.org)'
+
+      append :whitespace do
+        rule %r://.*$:, Comment
+
+        # comments are non-nesting, so a single regex will do
+        rule %r:/[*].*?[*]/:m, Comment
+      end
+
+      prepend :name do
+        rule Javascript.id_regex, Name::Label
+        rule %r/".*?"/, Name::Label
+      end
+
+      # comments can appear between keys and :, so we have to
+      # manage our states a little more carefully
+      append :object do
+        rule %r/:/ do
+          token Punctuation
+          goto :object_value
+        end
+      end
+
+      state :object_value do
+        mixin :value
+        rule %r/,/ do
+          token Punctuation
+          goto :object
+        end
+
+        rule %r/}/, Punctuation, :pop!
+      end
+
+      append :value do
+        rule %r/'/, Str::Single, :sstring
+      end
+
+      state :sstring do
+        rule %r/[^\\']+/, Str::Single
+        rule %r/\\./m, Str::Escape
+        rule %r/'/, Str::Single, :pop!
+      end
+
+      # can escape newlines
+      append :string do
+        rule %r/\\./m, Str::Escape
+      end
+
+      # override: numbers are very different in json5
+      state :constants do
+        rule %r/\b(?:true|false|null)\b/, Keyword::Constant
+        rule %r/[+-]?\b(?:Infinity|NaN)\b/, Keyword::Constant
+        rule %r/[+-]?0x\h+/i, Num::Hex
+
+        rule %r/[+-]?\d*\.\d*(?:e[+-]?\d+)?/i, Num::Float
```

Comment:
I think this rule matches everything that would have been matched by the below two rules

```ruby                           
rule %r/[+-]?[.]\d+/, Num::Float
rule %r/[+-]?\d+[.]/, Num::Float
```

so I took them out.

Reply:
@jneen Are you happy with this change?

Reply:
As long as the visual spec looks good! I would have put any edge cases in there.

Reply:
I should note that this matches just plain `.`, which I don't think is valid syntax

Reply:
Good point. Maybe we should put the two rules back and copy the exponent portion into them?

Reply:
you could avoid this case with `/(?:\d+[.]\d*|[.]\d+)/` which would be plenty performant i think.

Reply:
What happened to this? almost 5 years has progressed, I'm not really a Ruby dev but it would be nice to see this moving forward rather than stagnating.

Reply:
@robvdl Thank you for your patience. This has slipped my radar. I can take over and have look at it.

Reply:
I just came across it for the first time today :) it's fine.

I ran into it because Jetbrains IDE's render json5 while Gitlab didn't.